src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+};
+
+enum OrcCrate
+{
+    NPC_ORC_SEA_PUP                             = 44914
+};
+
+// 44915 - Orc Crate
+struct npc_silverpine_orc_crate : public ScriptedAI
+{
+    npc_silverpine_orc_crate(Creature* creature) : ScriptedAI(creature) { }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (summoner->GetEntry() == NPC_ORC_SEA_PUP)
+        {
+            me->SetUnitFlag(UNIT_FLAG_UNINTERACTIBLE);
+
+            me->EnterVehicle(summoner->ToCreature());
+        }
+    }
+};
+
+enum PickUpOrcCrate
+{
+    SPELL_KILL_CREDIT_SEA_DOG_CRATE             = 83843
+};
+
+// 83838 - Pick Up Orc Crate
+class spell_silverpine_pick_up_orc_crate : public SpellScript
+{
+    PrepareSpellScript(spell_silverpine_pick_up_orc_crate);
+
+    bool Validate(SpellInfo const* /*spellInfi*/) override
+    {
+        return ValidateSpellInfo(
+        {
+            SPELL_SUMMON_ORC_CRATE,
+            SPELL_KILL_CREDIT_SEA_DOG_CRATE
+        });
+    }
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        GetHitUnit()->CastSpell(nullptr, SPELL_SUMMON_ORC_CRATE);
+        GetHitUnit()->CastSpell(nullptr, SPELL_KILL_CREDIT_SEA_DOG_CRATE);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_silverpine_pick_up_orc_crate::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+};
+
+// 83840 - Despawn All Summons
+class spell_silverpine_despawn_all_summons_rear_guard : public SpellScript
+{
+    PrepareSpellScript(spell_silverpine_despawn_all_summons_rear_guard);
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+
+        if (Creature* target = GetHitCreature())
+        {
+            if (target->GetOwnerGUID() == caster->GetGUID())
+                target->DespawnOrUnsummon(4s);
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_silverpine_despawn_all_summons_rear_guard::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+};
+
+enum MutantBushChicken
+{
+    QUEST_POISONOUS_IF_YOU_INGEST_IT            = 27088,
+
+    NPC_FOREST_ETTIN                            = 44367,
+
+    EVENT_CHECK_ETTIN                           = 1,
+
+    ACTION_GRAB_CHICKEN                         = 1,
+
+    SOUND_CHICKEN_MOUNT_WOUND                   = 15936
+};
+
+// 44935 - Mutant Bush Chicken
+struct npc_silverpine_mutant_bush_chicken : public ScriptedAI
+{
+    npc_silverpine_mutant_bush_chicken(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);
+
+        _events.ScheduleEvent(EVENT_CHECK_ETTIN, 1s);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+        {
+            _playerGUID = player->GetGUID();
+
+            if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
```

Comment:
you already have access to `player` here from first condition, this condition is unnecessary

Reply:
Feedback applied.